prometheus-ksonnet/lib/kube-state-metrics.libsonnet

```diff
@@ -106,4 +106,50 @@
 
   kube_state_metrics_service:
     $.util.serviceFor($.kube_state_metrics_deployment),
+
+  prometheus_config+:: {
+    scrape_configs+: [
+      // A separate scrape config for kube-state-metrics which doesn't
+      // add namespace, container, and pod labels, instead taking
+      // those labels from the exported timeseries. This prevents them
+      // being renamed to exported_namespace etc.  and allows us to
+      // route alerts based on namespace and join KSM metrics with
+      // cAdvisor metrics.
+      {
+        job_name: '%s/kube-state-metrics' % $._config.namespace,
+        kubernetes_sd_configs: [{
+          role: 'pod',
+          namespaces: {
+            names: [$._config.namespace],
+          },
+        }],
+
+        relabel_configs: [
+
+          // Drop anything who's service is not kube-state-metrics
+          // Rename jobs to be <namespace>/<name, from pod name label>
```

Comment:
This line is a leftover and needs to go. (Again, was the case previously, but let's fix.)